src/osd/ECBackend.cc

```diff
@@ -1709,12 +1709,19 @@ void ECBackend::do_read_op(ReadOp &op)
     }
   }
 
+  bool should_read_local = false;
+  ECSubRead local_read_op;
```

Comment:
ECSubRead *local_read_op;
and drop the bool.  This way we avoid the copy?

Reply:
or put the ECSubReadReply up at this level with the bool, and do the handle_sub_read inside the loop.

Reply:
@liewegas   

drop the bool is ok，but i think do the handle_sub_read inside the loop, read may costs time(local read+remote read ); do the handle_sub_read outside the loop，it may costs time max(remote read , local read );i think it is better to do outside the loop.

like this:
    ECSubRead *local_read_op;

    ......

    if (i->first == get_parent()->whoami_shard()) {
      local_read_op = &(i->second);
      continue;
    } 

   .......

    dout(10) << __func__ << ": started " << op << dendl;
    if (local_read_op) {
      dout(10) << __func__ << ": starting local read " << op << dendl;
      ECSubReadReply reply;
      RecoveryMessages rm;
      handle_sub_read(get_parent()->whoami_shard(), *local_read_op, &reply, op.trace);    
      handle_sub_read_reply(get_parent()->whoami_shard(), reply, &rm, op.trace);
      dispatch_recovery_messages(rm, priority);
    }

Reply:
good point!